components/director/internal/domain/application/service.go

```diff
@@ -712,6 +716,141 @@ func (s *service) DeleteLabel(ctx context.Context, applicationID string, key str
+		return nil, errors.Wrapf(err, "while trying to merge applications with ids %s and %s", destID, srcID)
+	}
+
+	log.C(ctx).Infof("Merging labels for applications with ids %s and %s", destID, srcID)
+	destAppLabelsMerged, err := s.handleMergeLabels(srcAppLabels, destAppLabels)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while trying to merge labels for applications with ids %s and %s", destID, srcID)
+	}
+
+	log.C(ctx).Infof("Deleting source application with id %s", srcID)
+	if err := s.Delete(ctx, srcID); err != nil {
+		return nil, err
+	}
+
+	log.C(ctx).Infof("Updating destination app with id %s", srcID)
+	if err := s.appRepo.Update(ctx, appTenant, destApp); err != nil {
+		return nil, err
+	}
+
+	if err := s.labelUpsertService.UpsertMultipleLabels(ctx, appTenant, model.ApplicationLabelableObject, destID, destAppLabelsMerged); err != nil {
+		return nil, err
+	}
+
+	return s.appRepo.GetByID(ctx, appTenant, destID)
+}
+
+// handleMergeLabels merges source labels into destination labels
+func (s *service) handleMergeLabels(srcAppLabels, destAppLabels map[string]*model.Label) (map[string]interface{}, error) {
+	destIntSysID := fmt.Sprintf("%v", destAppLabels[intSysKey].Value)
+	srcIntSysID := fmt.Sprintf("%v", srcAppLabels[intSysKey].Value)
+	if len(destIntSysID) == 0 && len(srcIntSysID) > 0 {
+		destAppLabels[intSysKey].Value = srcIntSysID
+	}
+
+	srcScenarios, ok := srcAppLabels[model.ScenariosKey].Value.([]interface{})
+	if !ok {
+		return nil, errors.Errorf("error while converting source scenario with ID %s to slice", srcAppLabels[model.ScenariosKey].ID)
+	}
+
+	destScenarios, ok := destAppLabels[model.ScenariosKey].Value.([]interface{})
+	if !ok {
+		return nil, errors.Errorf("error while converting destination scenario with ID %s to slice", destAppLabels[model.ScenariosKey].ID)
+	}
+
+	srcScenariosStrSlice := make([]string, 0, len(srcScenarios))
+	for _, srcScenario := range srcScenarios {
+		srcScenarioStr, ok := srcScenario.(string)
+		if !ok {
+			return nil, errors.Errorf("error while converting source scenario to string")
+		}
+
+		srcScenariosStrSlice = append(srcScenariosStrSlice, srcScenarioStr)
+	}
+
+	destScenariosStrSlice := make([]string, 0, len(destScenarios))
+	for _, destScenario := range destScenarios {
+		destScenarioStr, ok := destScenario.(string)
+		if !ok {
+			return nil, errors.Errorf("error while converting destination scenario to string")
+		}
+
+		destScenariosStrSlice = append(destScenariosStrSlice, destScenarioStr)
+	}
+
+	for _, srcScenario := range srcScenariosStrSlice {
+		if !containsStr(destScenariosStrSlice, srcScenario) {
+			destScenariosStrSlice = append(destScenariosStrSlice, srcScenario)
+		}
+	}
+
+	if err := mergo.Merge(&destAppLabels, srcAppLabels); err != nil {
+		return nil, errors.Wrapf(err, "while trying to merge labels")
+	}
+
+	destAppLabels[model.ScenariosKey].Value = destScenariosStrSlice
+	destAppLabels[managedKey].Value = srcAppLabels[managedKey].Value
```

Comment:
Regarding the final value of the `managed` label:
- if `src` is auto-discovered and `dest` is customer-managed then we want the label `managed` to stay `true` -> you cover that here
- if `src` is customer-managed and `dest` is auto-discovered then we AGAIN want the label `managed` to stay `true`, because the final app will again have all fields (like systemNumber) which constitute an auto-discovered app (managed by system fetcher) -> this is not covered here and the final value for `managed` would actually be `false` with current logic.

Ultimately this will require just a slight tweak of the sort:
```
if destAppLabels[managedKey].Value.(bool) == true ||  srcAppLabels[managedKey].Value.(bool) == true {
  destAppLabels[managedKey].Value = true
}
```